server/ServerNetworking.cpp

```diff
@@ -167,7 +167,7 @@ PlayerConnection::PlayerConnection(boost::asio::io_context& io_context,
 
 PlayerConnection::~PlayerConnection() {
     boost::system::error_code error;
-    m_socket.shutdown(boost::asio::ip::tcp::socket::shutdown_both, error);
+    m_socket->shutdown(boost::asio::ip::tcp::socket::shutdown_both, error);
```

Comment:
Why is this done synchronously and not in the thread with the `close` call?

Reply:
Because I've been cautch freezes only on `close`